src/ORM/FieldType/DBForeignKey.php

```diff
@@ -87,12 +87,29 @@ public function scaffoldFormField($title = null, $params = null)
 
         // Add the count of the list to a cache for subsequent calls
         if (!isset(static::$foreignListCache[$hasOneClass])) {
+            // Let the DB do the threshold check as it will be faster - depending on the SQL engine it might only have
+            // to count indexes
+            $dataQuery = $list->dataQuery()->getFinalisedQuery();
+
+            $distinct = $dataQuery->getDistinct();
+            $count = '*';
+
+            if ($distinct) {
+                $dataQuery->setDistinct(false);
+                $count = 'DISTINCT *';
+            }
+
+            $dataQuery->setOrderBy(false);
+            $dataQuery->setSelect(['over_threshold' => 'count('. $count . ') > ' . (int) $threshold]);
+            $result = $dataQuery->execute()->column('over_threshold')[0];
             static::$foreignListCache[$hasOneClass] = [
-                'count' => $list->count(),
+                'overThreshold' => is_bool($result) ? $result : ($result === '1' || $result === 't'),
```

Comment:
Worth adding a note that the `1` `t` checks are for B/C with Postgres < 2.2?

Reply:
Yep I'll do that.